go/libkb/session.go

```diff
@@ -285,13 +287,29 @@ func (s *Session) check() error {
 		}
 	} else {
 		s.G().Log.Notice("Stored session expired")
-		s.valid = false
+		s.Invalidate()
 	}
 
 	s.G().Log.Debug("- Checked session")
 	return nil
 }
 
+// Invalidate marks the session as invalid and posts a logout
+// notification.
+func (s *Session) Invalidate() {
+	s.G().Log.Debug("invalidating session")
+	s.clearState()
+	s.G().NotifyRouter.HandleLogout()
```

Comment:
Trying to track it down.  Maybe not? We really need to clean this stuff up, there are so many layers of indirection.  This feel very unmaintainable.  We're bound to introduce code in the future that either drops the notification or calls it twice.